### PR TITLE
CV-1099 Restructure app template for simpler integration

### PR DESCRIPTION
**Pull request by Saravana-Gunaseelan**
Also added a example for saving multiview images.
Will add Unit test to this PR once the functions added are approved.

### CODE REVIEW

File: apptemplate/src/androidTest/java/com/leia/apptemplate/BaseActivityTest.java [hunk 1]
```diff
@@ -20,12 +20,12 @@
...
     public void saveDefaultImageToFile_correctResourceID_returnsUri() {
-        Uri uri = activityActivityTestRule.getActivity().saveDefaultImageToFile(R.raw.test_img);
+        Uri uri = activityActivityTestRule.getActivity().saveResourceToFile(R.raw.test_img);
         assertEquals("/storage/emulated/0/Android/data/com.leia.apptemplate.test/cache/test_img.jpg", uri.getPath());
     }
 
     @Test(expected = Resources.NotFoundException.class)
     public void saveDefaultImageToFile_incorrectResourceID_returnsUri() {
-        Uri uri = activityActivityTestRule.getActivity().saveDefaultImageToFile(1);
+        Uri uri = activityActivityTestRule.getActivity().saveResourceToFile(1);
```
Comment (PuneetKohli) [inline apptemplate/src/androidTest/java/com/leia/apptemplate/BaseActivityTest.java]: What is `1` here?

File: apptemplate/src/main/java/com/leia/apptemplate/BaseActivity.java [hunk 2]
```diff
@@ -207,4 +249,59 @@ private String getFileNameFromResource(@RawRes int resource) {
...
         return fileNameArray[fileNameArray.length-1];
     }
+
+    /**
+     * Saves a multiviewImage into a temp file in the cache storage.
+     *
+     * @param multiviewImage multiviewImage to be saved
+     * @return URI of the saved file
+     */
+    public Uri saveMultiviewImageToFile(MultiviewImage multiviewImage) {
```
Comment (PuneetKohli) [inline apptemplate/src/main/java/com/leia/apptemplate/BaseActivity.java]: Should we be explicit and say `saveMultiviewImageToTempFile`?

File: apptemplate/src/androidTest/java/com/leia/apptemplate/BaseActivityTest.java (hunk 1 above)
Comment (Saravana-Gunaseelan) [reply apptemplate/src/androidTest/java/com/leia/apptemplate/BaseActivityTest.java]: an invalid resource id

File: apptemplate/src/main/java/com/leia/apptemplate/BaseActivity.java (hunk 2 above)
Comment (Saravana-Gunaseelan) [reply apptemplate/src/main/java/com/leia/apptemplate/BaseActivity.java]: will change it. 
Do you think I should move these functions 
`saveMultiviewImageToTempFile` and `saveResourceToFile` to a Util class.

**Pull request merged**